slack-plugin/src/main/kotlin/slack/gradle/lint/LintTasks.kt

```diff
@@ -87,139 +255,196 @@ internal object LintTasks {
+          // Lint doesn't understand AppComponentFactory
+          // https://issuetracker.google.com/issues/243267012
+          disable += "Instantiatable"
         }
       }
-      return // nothing else to do!
+
+      lintConfig = project.rootProject.layout.projectDirectory.file("config/lint/lint.xml").asFile
+
+      baseline =
+        project.objects
+          .fileProperty()
+          .fileValue(File(project.projectDir, slackProperties.lintBaselineFileName))
+          .get()
+          .asFile
     }
+  }
 
-    val componentsExtension =
-      when (extension) {
-        is AppExtension -> {
-          project.extensions.getByType<ApplicationAndroidComponentsExtension>()
-        }
-        is LibraryExtension -> {
-          project.extensions.getByType<LibraryAndroidComponentsExtension>()
-        }
-        is TestExtension -> {
-          project.extensions.getByType<TestAndroidComponentsExtension>()
-        }
-        else -> error("No AndroidComponentsExtension found for project ${project.path}")
+  /**
+   * If the project is using multiplatform, adds configurations and source sets expected by the lint
+   * plugin, which allows it to configure itself when running against a non-Android multiplatform
+   * project.
+   *
+   * The version of lint that we're using does not directly support Kotlin multiplatform, but we can
+   * synthesize the necessary configurations and source sets from existing `jvm` configurations and
+   * `kotlinSourceSets`, respectively.
+   *
+   * This method *must* run after evaluation.
+   */
+  private fun Project.addSourceSetsForMultiplatformAfterEvaluate() {
+    val kmpTargets = project.multiplatformExtension?.targets ?: return
+
+    // Synthesize target configurations based on multiplatform configurations.
+    val kmpApiElements = kmpTargets.map { it.apiElementsConfigurationName }
+    val kmpRuntimeElements = kmpTargets.map { it.runtimeElementsConfigurationName }
+    listOf(kmpRuntimeElements to "runtimeElements", kmpApiElements to "apiElements").forEach {
+      (kmpConfigNames, targetConfigName) ->
+      project.configurations.maybeCreate(targetConfigName).apply {
+        kmpConfigNames
+          .mapNotNull { configName -> project.configurations.findByName(configName) }
+          .forEach { config -> extendsFrom(config) }
       }
-    componentsExtension.onVariants { variant ->
-      val lintTaskName = "lint${variant.name.capitalizeUS()}"
-      project.log("Adding $lintTaskName to ciLint task")
-      ciLintTask.configure {
-        // Even if the task isn't created yet, we can do this by name alone and it will resolve at
-        // task configuration time.
-        dependsOn(lintTaskName)
+    }
+
+    // Synthesize source sets based on multiplatform source sets.
+    val javaExtension =
+      project.extensions.findByType(JavaPluginExtension::class.java)
+        ?: throw GradleException("Failed to find extension of type 'JavaPluginExtension'")
```

Comment:
if you're going to throw anyway, why not just use `getByType()`? Is `java` guaranteed to be present anyway if you're here?

Reply:
[5c04028](https://github.com/slackhq/slack-gradle-plugin/pull/612/commits/5c040286346d6a49cb24346a54c569baf5e9ae91)